src/ice_spi.c

```diff
@@ -22,28 +22,33 @@
  * SOFTWARE.
  */
 
-// TODO: For now, this is a bit-banged library which is sub-optimal.
-// We could use the hardware SPI for when in forward pin order, and
-// a PIO state machine for reverse pin order. But since we are going
-// to need the reverse pin order anyway, we might as well make use
-// of the PIO state machine right away. This way only one PIO state
-// machine is used and no extra SPI peripheral.
-//
-// So the next evolution of this driver is to use PIO, and after that
-// DMA channels along with PIO.
-
 #include <stddef.h>
 #include <stdint.h>
 #include <stdbool.h>
 #include <assert.h>
 #include "hardware/sync.h"
 #include "pico/stdlib.h"
+#include "hardware/spi.h"
+#include "hardware/dma.h"
 #include "boards/pico_ice.h"
 #include "ice_spi.h"
 
 volatile static bool g_transfer_done = true;
 volatile static void (*g_async_callback)(volatile void *);
 volatile static void *g_async_context;
+static int dma_tx, dma_rx;
+static uint32_t dma_word;
+
+static void spi_irq_handler(void) {
+    if (dma_hw->ints1 & (1u << dma_rx))
```

Comment:
It looks like there is no wrapper for it, and that it is how Raspberry Pi does it too...
https://github.com/raspberrypi/pico-examples/blob/eca13acf57916a0bd5961028314006983894fc84/pio/ws2812/ws2812_parallel.c#L229-L237

I'll add a comment later.

Reply:
I hadn't realised, but actually there are API functions for this, which were used in `ice_comm`